checker/src/org/checkerframework/checker/lock/LockStore.java

```diff
@@ -137,4 +159,78 @@ protected void internalDotOutput(StringBuilder result) {
                 + "\\n");
         super.internalDotOutput(result);
     }
+
+    @Override
+    protected boolean isSideEffectFree(AnnotatedTypeFactory atypeFactory,
+            ExecutableElement method) {
+        return ((LockAnnotatedTypeFactory) atypeFactory).methodSideEffectAnnotation(method, false) == SideEffectAnnotation.RELEASESNOLOCKS ||
+               super.isSideEffectFree(atypeFactory, method);
+    }
+
+    // Same as super method, except for localVariableValues.clear();
```

Comment:
Is this related to #524 ? Could you please add a comment explaining why the local variables need to be cleared?